crates/revm/src/db/emptydb.rs

```diff
@@ -1,80 +1,87 @@
 use core::{convert::Infallible, marker::PhantomData};
 use revm_interpreter::primitives::{
     db::{Database, DatabaseRef},
-    keccak256, AccountInfo, Bytecode, B160, B256, U256,
+    AccountInfo, Bytecode, B160, B256, U256,
 };
 
+/// An empty database that always returns default values when queried.
 pub type EmptyDB = EmptyDBTyped<Infallible>;
 
-impl Default for EmptyDB {
-    fn default() -> Self {
-        Self {
-            keccak_block_hash: false,
-            _phantom: PhantomData,
-        }
-    }
-}
-
 /// An empty database that always returns default values when queried.
+///
+/// This is generic over a type which is used as the database error type.
 #[derive(Debug, Clone)]
-pub struct EmptyDBTyped<T> {
-    pub keccak_block_hash: bool,
-    pub _phantom: PhantomData<T>,
+pub struct EmptyDBTyped<E> {
+    _phantom: PhantomData<E>,
+}
+
+// Don't derive it because it doesn't need `E: Default`.
+impl<E> Default for EmptyDBTyped<E> {
+    #[inline(always)]
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
-impl<T> EmptyDBTyped<T> {
+impl<E> EmptyDBTyped<E> {
+    #[inline(always)]
     pub fn new() -> Self {
         Self {
-            keccak_block_hash: false,
             _phantom: PhantomData,
         }
     }
 
+    #[deprecated = "use `new` instead"]
+    #[inline(always)]
     pub fn new_keccak_block_hash() -> Self {
-        Self {
-            keccak_block_hash: true,
-            _phantom: PhantomData,
-        }
+        Self::new()
     }
 }
 
-impl<T> Database for EmptyDBTyped<T> {
-    type Error = T;
+impl<E> Database for EmptyDBTyped<E> {
+    type Error = E;
 
+    #[inline(always)]
     fn basic(&mut self, address: B160) -> Result<Option<AccountInfo>, Self::Error> {
         <Self as DatabaseRef>::basic(self, address)
     }
 
+    #[inline(always)]
     fn code_by_hash(&mut self, code_hash: B256) -> Result<Bytecode, Self::Error> {
         <Self as DatabaseRef>::code_by_hash(self, code_hash)
     }
 
+    #[inline(always)]
     fn storage(&mut self, address: B160, index: U256) -> Result<U256, Self::Error> {
         <Self as DatabaseRef>::storage(self, address, index)
     }
 
+    #[inline(always)]
     fn block_hash(&mut self, number: U256) -> Result<B256, Self::Error> {
         <Self as DatabaseRef>::block_hash(self, number)
     }
 }
 
-impl<T> DatabaseRef for EmptyDBTyped<T> {
-    type Error = T;
-    /// Get basic account information.
+impl<E> DatabaseRef for EmptyDBTyped<E> {
+    type Error = E;
+
+    #[inline(always)]
     fn basic(&self, _address: B160) -> Result<Option<AccountInfo>, Self::Error> {
         Ok(None)
     }
-    /// Get account code by its hash
+
+    #[inline(always)]
     fn code_by_hash(&self, _code_hash: B256) -> Result<Bytecode, Self::Error> {
         Ok(Bytecode::new())
     }
-    /// Get storage value of address at index.
```

Comment:
Docs on `impl Trait` functions are unnecessary if they are the same as the trait's.